traversal/procedure/ProcedureEdge.java

```diff
@@ -327,9 +328,13 @@ private Isa(VERTEX_ISA_FROM from, VERTEX_ISA_TO to, int order,
                 this.isTransitive = isTransitive;
             }
 
-            FunctionalIterator<TypeVertex> isaTypes(ThingVertex thing) {
-                if (!isTransitive) return single(thing.type());
-                else return loop(thing.type(), Objects::nonNull, v -> v.outs().edge(SUB).to().firstOrNull());
+            Seekable<TypeVertex, Order.Asc> isaTypes(ThingVertex thing) {
+                if (!isTransitive) return iterateSorted(ASC, thing.type());
+                else {
+                    TreeSet<TypeVertex> superTypes = new TreeSet<>();
+                    loop(thing.type(), Objects::nonNull, v -> v.outs().edge(SUB).to().firstOrNull()).forEachRemaining(superTypes::add);
+                    return iterateSorted(ASC, superTypes);
```

Comment:
note: this could be the souce of the performance regression
this would be unavoidable though!